lib/rubocop/cop/rspec/expect_change.rb

```diff
@@ -0,0 +1,102 @@
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    module RSpec
+      # Checks for consistent style of change matcher.
+      #
+      # Enforces either `and_return` or block-style return in the cases
+      # where the returned value is constant. Ignores dynamic returned values
+      # are the result would be different
+      #
+      # This cop can be configured using the `EnforcedStyle` option
+      #
+      # @example `EnforcedStyle: block`
+      #   # bad
+      #   expect(run).to change(Foo, :bar)
+      #
+      #   # good
+      #   expect(run).to change { Foo.bar }
+      #
+      # @example `EnforcedStyle: method_call`
+      #   # bad
+      #   expect(run).to change { Foo.bar }
+      #   expect(run).to change { foo.baz }
+      #
+      #   # good
+      #   expect(run).to change(Foo, :bar)
+      #   expect(run).to change(foo, :baz)
+      #   # also good when there are arguments or chained method calls
+      #   expect(run).to change { Foo.bar(:count) }
+      #   expect(run).to change { user.reload.name }
+      #
+      class ExpectChange < Cop
+        include ConfigurableEnforcedStyle
+
+        MSG_BLOCK = 'Prefer `change(%<obj>s, :%<attr>s)`.'.freeze
+        MSG_CALL = 'Prefer `change { %<obj>s.%<attr>s }`.'.freeze
+
+        def_node_matcher :expect_change_with_block, <<-PATTERN
+          (block
+            (send nil? :change)
+            (args)
+            (send ({const send} nil? $_) $_)
+          )
+        PATTERN
+
+        def_node_matcher :expect_change_with_arguments, <<-PATTERN
```

Comment:
Pardon my OCD, but could you define this method before defining `expect_change_with_block`? That is the order in which the methods are used throughout the class.

Reply:
I value your OCD.